lab/dp/problems/sum_query/main.cpp

```diff
@@ -0,0 +1,150 @@
+// Copyright 2020 Nichita Radu <radunichita99@gmail.com>
+
+// SumQuery - Programare dinamica
+// Complexitate O(M * N)
+
+// https://leetcode.com/problems/range-sum-query-2d-immutable/
+
+#include <bits/stdc++.h>
+
+class Task {
+    // N si M sunt dimensiunile matricei, K e numarul de query-uri
+    int N, M, K;
+
+    // matricea de elemente
+    std::vector<std::vector<int>> matrix;
+
+    // query-urile de forma <(start_x, start,y), (finish_x), (finish_y)>.
+    std::vector<std::pair<std::pair<int, int>, std::pair<int, int>>> queries;
+
+    // pe pozitia [i][j] am suma elementelor submatricei (1,1) pana la (i, j).
+    std::vector<std::vector<int>> dp;
+
+    // vectorul se solutii
+    std::vector<int> solution;
+
+    void read_input() {
+        std::cin >> N >> M >> K;
+        matrix = std::vector<std::vector<int>>(N + 1, std::vector<int>(M + 1, 0));
+        dp     = std::vector<std::vector<int>>(N + 1, std::vector<int>(M + 1, 0));
+
+        for (int i = 1; i <= M; i++) {
+            for (int j = 1; j <= N; j++) {
+                std::cin >> matrix[i][j];
+            }
+        }
+
+        queries = std::vector<std::pair<std::pair<int, int>, std::pair<int, int>>>(K);
```

Comment:
I found this very confusion. Please use `push_back` in the loop and avoid this ugly code.

`queries.push_back({ {x, y}, {p, q} };` - this whould make the code mode readable as you don't need the line 37.